### PR TITLE
Remove document types that are no longer used

### PR DESCRIPTION
**Pull request by kevindew**
This is a piece of work following the discovery into content-types, where there was an outlining of document types that could be removed: https://docs.google.com/document/d/1JnT6ZAM_Pn8L4s3eXsyfR2SAWyHwVpMBJ9V-T0EDq6Y/edit#

I've been through that list and checked with Publishing API to remove ones that are no longer needed.

The document types I've removed are:

- calculator
- coming_soon
- detailed_guidance
- dfid_research_output
- imported
- knowledge_alpha
- placeholder_ministerial_role
- placeholder_organisation
- placeholder_person
- placeholder_policy_area
- placeholder_worldwide_organisation
- policy_area
- unpublishing
- news_article

Of the above document I didn't remove the following ones:

- facet_group (already removed)
- facet_value (already removed)
- person (doc suggests removing placeholder form but that's already forbidden)
- placeholder (I believe still used, can be removed when Whitehall frontend migration is complete)
- publication (didn't seem to exist)
- special_route (doc suggests removing placeholder form but that's already forbidden)
- topical_event (doc suggests removing placeholder form but that's already forbidden)
- world_location (doc suggests removing placeholder form but that's already forbidden)
- world_location_news_article (didn't seem to exist)

This is easiest reviewed commit by commit.

---

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

This application is owned by the publishing platform team. Please let us know in #govuk-publishing-platform when you raise any PRs.

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: content_schemas/allowed_document_types.yml [hunk 1]
```diff
@@ -10,7 +10,6 @@
 - authored_article
 - board_member_role
 - business_finance_support_scheme
-- calculator
```
Comment (chrislo) [inline content_schemas/allowed_document_types.yml:13]: Can the corresponding `.jsonnet` (i.e. https://github.com/alphagov/publishing-api/blob/main/content_schemas/formats/calendar.jsonnet) also be removed in this commit?

Comment (chrislo) [reply content_schemas/allowed_document_types.yml:13]: I think this comment also applies to the other content types that are being removed, but I might be missing some context!

Comment (kevindew) [reply content_schemas/allowed_document_types.yml:13]: Yup, I've tried to remove all the corresponding jsonnet files with them. I think you've just made a minor error with this catch as you've pulled out the calendar.jsonnet file not calculator.jsonnet.

Thanks for taking a look.

**Review by brucebolt — APPROVED** [review]
Looks good to me.  Thanks for cleaning this up!

**Pull request merged**